build.gradle

```diff
@@ -205,11 +202,11 @@ configure(publishProjects) { Project subproject ->
 
 artifactory {
     publish {
-        contextUrl = 'https://software.r3.com/artifactory'
+        contextUrl = artifactory_contextUrl
         repository {
             repoKey = 'corda-dev'
-            username = System.getenv('CORDA_ARTIFACTORY_USERNAME') ?: System.getProperty('corda.artifactory.username')
-            password = System.getenv('CORDA_ARTIFACTORY_PASSWORD') ?: System.getProperty('corda.artifactory.password')
+            username = project.findProperty('cordaArtifactoryUsername') ?: System.getenv('CORDA_ARTIFACTORY_USERNAME') ?: System.getProperty('corda.artifactory.username')
+            password = project.findProperty('cordaArtifactoryPassword') ?: System.getenv('CORDA_ARTIFACTORY_PASSWORD') ?: System.getProperty('corda.artifactory.password')
```

Comment:
I'm wondering maybe we can avoid having this check. My understanding of the `findProperty` procedure is that it will look up for some property set either in this file or in the `settings` file. This probably that the username and password may be hardcoded in either of those two files, which could be rarely the case.

Reply:
This check allows people (i.e. "me" :wink:) to store my Artifactory API key in my `$HOME/.gradle/gradle.properties` file, and hence publish `SNAPSHOT` builds without needing either to set environment variables or Java system properties.

Putting my credentials into `settings.gradle` or `build.gradle` would be Bad Idea :tm:.